internal/libs/clist/clist_property_test.go

```diff
@@ -0,0 +1,56 @@
+package clist_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+
+	"github.com/tendermint/tendermint/internal/libs/clist"
+)
+
+func TestCListProperties(t *testing.T) {
+	rapid.Check(t, rapid.Run(&clistModel{}))
+}
+
+type clistModel struct {
+	clist *clist.CList
+
+	state []*clist.CElement
```

Comment:
I'm assuming this is meant to be a "model" of the correct behavior. Maybe it could be named to indicate that?